### PR TITLE
paginate collect rewards

### PR DESCRIPTION
**Pull request by magiodev**
## 4. Checklist

- [ ] Does the Readme need to be updated?

### CODE REVIEW

**Review by LaurensKubat — COMMENTED** [review]
one small comment but in general the draft lgtm

File: smart-contracts/contracts/cl-vault/src/instantiate.rs [hunk 1]
```diff
@@ -58,9 +59,11 @@ pub fn handle_instantiate(
         },
     )?;
 
+    IS_COLLECTING.save(deps.storage, &false)?;
```
[the remark this answers is not in the document]
Comment (magiodev) [reply smart-contracts/contracts/cl-vault/src/instantiate.rs]: done! everything has been polished and new tests have been pushed, previous one adjusted and a new test case for depositing in between of collection too

**Review by ajansari95 — APPROVED** [review]
lgtm

**Review by LaurensKubat — COMMENTED** [review]
small comment around code cleanliness, but lgtm in general

File: smart-contracts/contracts/cl-vault/src/rewards/distribution.rs [hunk 2]
```diff
@@ -141,20 +189,12 @@ pub fn execute_distribute_rewards(
     _env: Env,
     amount_of_users: Uint128,
 ) -> Result<Response, ContractError> {
-    let is_distributing = IS_DISTRIBUTING.may_load(deps.storage)?.unwrap();
-    if !is_distributing {
+    let rewards_status = REWARDS_STATUS.load(deps.storage).unwrap();
+    if rewards_status != RewardsStatus::Distributing {
```
[the remark this answers is not in the document]
Comment (magiodev) [reply smart-contracts/contracts/cl-vault/src/rewards/distribution.rs:193]: Sure, i'll leave this as if, but i noticed a redeclaration wasn't necessary before computing the last iteration. Other comments are solved now.

**Review by LaurensKubat — APPROVED** [review]
lgtm

**Review by ajansari95 — APPROVED** [review]
LGTM

**Pull request merged**